packages/components/src/truncate/hook.ts

```diff
@@ -33,17 +33,29 @@ export default function useTruncate(
 
 	const cx = useCx();
 
-	const truncatedContent = truncateContent(
-		typeof children === 'string' ? children : '',
-		{
-			ellipsis,
-			ellipsizeMode,
-			limit,
-			numberOfLines,
-		}
-	);
+	let childrenAsText;
+	if ( typeof children === 'string' ) {
+		childrenAsText = children;
+	} else if ( typeof children === 'number' ) {
+		childrenAsText = children.toString();
+	} else {
+		// eslint-disable-next-line no-console
+		console.warn(
+			`Truncate: text truncation has been disabled, since it is only available when passing 'children' of type 'string' or 'number'`
```

Comment:
```suggestion
			`wp.components.Truncate: text truncation has been disabled, since it is only available when passing 'children' of type 'string' or 'number'`
```

Probably better for clarity, since `Truncate` is used under the hood for a handful of more common components and the consumer may not be aware of what "Truncate" is specifically.